src/main/java/org/sopt/app/presentation/stamp/StampController.java

```diff
@@ -1,110 +1,93 @@
+    private final S3Service s3Service;
 
-    return new ResponseEntity<>(uploadStamp, getSuccessHeaders(), HttpStatus.OK);
-  }
+    private final StampResponseMapper stampResponseMapper;
 
-
-  @PutMapping("/{missionId}")
-  public ResponseEntity<?> editStamp(
-      @RequestHeader("userId") String userId,
-      @PathVariable Long missionId,
-      @RequestPart(value = "stampContent", required = false) StampRequestDto stampRequestDto,
-      @RequestPart(name = "imgUrl", required = false) List<MultipartFile> multipartFiles
-
-  ) {
-    //MultipartFile을 리스트에 넣어줬기 때문에 List 내부의 이미지파일에 isEmpty()를 적용해야 한다.
-    int checkNum = 1;
-    for (MultipartFile image : multipartFiles) {
-      if (image.isEmpty()) {
-        checkNum = 0;
-      }
+    @Operation(summary = "스탬프 조회하기")
+    @GetMapping("/{missionId}")
+    public ResponseEntity<StampResponse.Main> findStampByMissionAndUserId(
+            @RequestHeader("userId") String userId, @PathVariable Long missionId) {
+        val result = stampService.findStamp(userId, missionId);
+        val response = stampResponseMapper.of(result);
+        return ResponseEntity.status(HttpStatus.OK).body(response);
     }
 
-    StampResponseDto result = StampResponseDto.builder().build();
-    if (checkNum == 0) {
-
-      Stamp stamp = stampService.editStampContents(stampRequestDto, userId, missionId);
-      result.setStampId(stamp.getId());
-
-    } else {
-
-      List<String> imgPaths = s3Service.upload(multipartFiles);
-      Stamp uploadStamp = stampService.editStampWithImg(stampRequestDto,
-          imgPaths, userId, missionId);
+    @Operation(summary = "스탬프 등록하기")
+    @PostMapping("/{missionId}")
+    public ResponseEntity<StampResponse.Main> uploadStamp(
+            @RequestHeader("userId") String userId,
+            @PathVariable Long missionId,
+            @RequestPart("stampContent") StampRequest.RegisterStampRequest registerStampRequest,
+            @RequestPart(name = "imgUrl", required = false) List<MultipartFile> multipartFiles
+    ) {
+        stampService.checkDuplicateStamp(userId, missionId); // 스탬프 중복 검사체크
+        val imgPaths = s3Service.upload(multipartFiles);
+        val result = stampService.uploadStamp(registerStampRequest, imgPaths, userId, missionId);
+        val response = stampResponseMapper.of(result);
+        return ResponseEntity.status(HttpStatus.OK).body(response);
+    }
 
-      result.setStampId(uploadStamp.getId());
+    @Operation(summary = "스탬프 수정하기")
+    @PutMapping("/{missionId}")
+    public ResponseEntity<StampResponse.Id> editStamp(
+            @RequestHeader("userId") String userId,
+            @PathVariable Long missionId,
+            @RequestPart(value = "stampContent", required = false) StampRequest.EditStampRequest editStampRequest,
+            @RequestPart(name = "imgUrl", required = false) List<MultipartFile> multipartFiles
+
+    ) {
+        if (multipartFiles == null || multipartFiles.get(0).isEmpty()) {
```

Comment:
```suggestion
       val isEmptyFileList = (multipartFiles == null || multipartFiles.get(0).isEmpty())
        if (isEmptyFileList) {
```
이런 식으로 조건문을 읽기 쉬운 단어로 바꿔주면 가독성에 굉장히 도움이 될 거 같습니다!